conformance/lang/expressions/list_constructor.balt

```diff
@@ -0,0 +1,1459 @@
+             that has its read-only bit on, for array-typed const.
+Fail-Issue: ballerina-platform/ballerina-lang#13944
+Labels: array-type, is-expr, list-constructor-expr, module-const-decl, readonly-type, string
+
+const string[] strArr = [];
+const int[] intArr = [1, 2, 3, 4];
+
+function init() {
+    io:println(strArr is readonly); // @output true
+    io:println(intArr is readonly); // @output true
+}
+
+Test-Case: output
+Description: Test list-constructor-expr occurring within a const-expr will construct a structural value
+             that has its read-only bit on, for tuple-typed const.
+Fail-Issue: ballerina-platform/ballerina-lang#13944
+Labels: boolean, is-expr, list-constructor-expr, module-const-decl, readonly-type, string, tuple-type
+
+const [int, string, boolean] tuple1 = [11, "foo", true];
+const [int, [boolean, string], boolean] tuple2 = [11, [true, "foo"], true];
+
+function init() {
+    io:println(tuple1 is readonly); // @output true
+    io:println(tuple2 is readonly); // @output true
+}
+
+Test-Case: output
+Description: Test list-constructor-expr occurring within a const-expr will construct a structural value
+             that has its read-only bit on, for map-typed const.
+Fail-Issue: ballerina-platform/ballerina-lang#13944
+Labels: array-type, is-expr, list-constructor-expr, map-type, mapping-constructor-expr, member-access-expr,
+        module-const-decl, readonly-type, string
+
+const map<int[]> keyValues = {even: [2, 4, 6], odd: [1, 3, 5]};
+
+function init() {
+    io:println(keyValues["odd"] is readonly); // @output true
+    io:println(keyValues["even"] is readonly); // @output true
+}
+
+Test-Case: output
+Description: Test if there is a contextually expected type, then the inherent type of the newly created list
+             is derived from the applicable contextually expected type, for array.
+Labels: array-type, byte, is-expr, list-constructor-expr, singleton-type, union-type
+
+function init() {
+    int[] a1 = [1, 2, 3];
+    io:println(a1 is int[]); // @output true
+    io:println(a1 is byte[]); // @output false
+
+    byte[] a2 = [1, 2, 3];
+    io:println(a2 is byte[]); // @output true
+    io:println(a2 is (1|2|3)[]); // @output false
+}
+
+Test-Case: output
+Description: Test if there is a contextually expected type, then the inherent type of the newly created list
+             is derived from the applicable contextually expected type, for tuple.
+Labels: byte, is-expr, list-constructor-expr, singleton-type, tuple-type
+
+function init() {
+    [int, byte, 3] t1 = [1, 2, 3];
+    io:println(t1 is [int, byte, 3]); // @output true
+    io:println(t1 is [int, 2, 3]); // @output false
+}
+
+Test-Case: output
+Description: Test inherent type of the created list when the applicable contextually expected type
+             is a array-type-descriptor of (any|error).
+Fail-Issue: ballerina-platform/ballerina-lang#33377
```

Comment:
This issue has been fixed? 
But line number 166 has an `ambiguous type` error

Reply:
enabled. can you check now?